src/sentry/eventstream/kafka/backend.py

```diff
@@ -248,6 +271,29 @@ def commit_offsets():
                 )
                 commit(offsets_to_commit)
 
+        def process_message(message: Message) -> None:
+            task_kwargs = self._get_task_kwargs(message)
+            if not task_kwargs:
+                return
+
+            self.record_metrics(message.partition(), task_kwargs)
+            self._pending_futures.append(
+                executor.submit(self._dispatch_post_process_group_task, **task_kwargs)
+            )
+
+        def collect_results() -> None:
+            """
+            Collect results from all the pending futures. If any of the futures raised an exception,
+            we want to raise it so that the kafka offsets of the current batch don't get committed.
+            The batching is either all or none.
+            TODO: Add dead letter queue to handle this case.
```

Comment:
Is this the same behavior as today?

Reply:
Yes. Its the same behavior as today. If any exception is thrown during processing, the main loop would throw the exception and the offsets don't get commited.